runtime/v2/runc/container.go

```diff
@@ -49,7 +49,7 @@ func NewContainer(ctx context.Context, platform stdio.Platform, r *task.CreateTa
 	}
 
 	var opts options.Options
-	if r.Options != nil {
+	if r.Options != nil && r.Options.GetTypeUrl() != "" {
```

Comment:
I'm unsure this is the right place to fix this issue. Shouldn't it be handled by `typeurl.UnmarshalAny()`?

Reply:
@crosbymichael can you provide any insight here?

Reply:
Ya, i think this is fine.  It should be handled by the caller because of the `nil` check.  I'll have to think a bit on a better way to fix this in the lib but it shouldn't be common.